build/ci_build

```diff
@@ -119,6 +122,15 @@ def build_dockers(
             "--tag=%s" % tag,
         ]
 
+        if os.path.isdir(therock_path):
+            cmd.append("--build-arg=THEROCK_PATH=/tmp/therock/")
+            cmd.append("--build-context=therock=%s" % therock_path)
+        else:
+            if therock_path:
+                cmd.append("--build-arg=THEROCK_PATH=%s" % therock_path)
+            os.makedirs("tmp_therock", exist_ok=True)
+            cmd.append("--build-context=therock=tmp_therock")
```

Comment:
nit: prefer `pathlib` (https://docs.python.org/3/library/pathlib.html) over `os.path` for new code, especially if the existing code doesn't already heavily use `os.path`. Pathlib has typed objects for paths instead of raw strings so it is safer to use and more portable. There is a table at https://docs.python.org/3/library/pathlib.html#corresponding-tools showing which functions are similar.

You could also use a portable tmpdir name instead of `/tmp/therock` for eventual Windows support. For example: https://docs.python.org/3/library/tempfile.html or a path under $HOME using https://docs.python.org/3/library/pathlib.html#pathlib.Path.home.